models/iosxr/ospfv2/iosxr_ospfv2.yml

```diff
@@ -0,0 +1,1983 @@
+                  type: list
+                  elements: dict
+                  required_together:[[value,source,wildcard]]
+                  suboptions:
+                    value:
+                      description: Distance value (1-255).
+                      type: int
+                    source:
+                      description: Source IP address.
+                      type: str
+                    wildcard:
+                      description: IP wild card bits (A.B.C.D).
+                      type: str
+                    access_list:
+                      description: Access list name.
+                      type: str
+                ospf_distance:
+                  description: OSPF administrative distance.
+                  type: dict
+                  suboptions:
+                    external: 
+                      description: Distance for external routes.
+                      type: int
+                    inter_area:
+                      description: Distance for inter-area routes.
+                      type: int
+                    intra_area:
+                      description: Distance for intra-area routes.
+                      type: int 
+            distribute_link_state:
+              description: Enable Distribution of LSAs to external services.
+              type: dict
+              suboptions:
+                instance_id:
+                  description: Set distribution process instance identifier <0-4294967295>.
+                  type: int
+                throttle:
+                  description: Throttle time between successive LSA updates (<1-3600> seconds).
+                  type: int
+            distribute_list:
+              description: Filter networks in routing updates.
+              type: dict
+              suboptions:
+                access_list:
+                  description: Inbound/outbound access-list.
+                  type: dict
+                  required_together:[[name,direction]]
+                  suboptions:
+                    name:
+                      description: Access-list name
+                      type: str
+                    direction:
+                      description: Filter incoming/outgoing routing updates
+                      type: str
+                      choices:
+                        - in
+                        - out
+                    options:
+                      description: Specify additional parameters for outgoing updates only.
+                      type: dict
+                      suboptions:
+                        route_type:
+                          description: Type of routes.
+                          type: str
+                          choices: 
+                            - bgp
+                            - connected
+                            - dagr
+                            - ospf
+                            - static
+                        as_number: 
```

Comment:
It would probably be simpler to have a key `id` that addresses both bgp and the other protocols

Reply:
Yeah, that makes better sense